packages/block-editor/src/hooks/typography.js

```diff
@@ -48,33 +52,59 @@ function TypographyInspectorControl( { children } ) {
 	);
 }
 
+function useBLockSettings( name ) {
```

Comment:
While right now, this function is "adhoc" in the "typography" hook. I believe this can be one big unique function that "builds" this object per block and make it available for all hooks... in the `BlockEditContext` or something like that.